File: UI/tendergo_desktop/lib/shared/models/requests/paged_search_request.dart
class PagedSearchRequest {
  final int page;
  final int pageSize;
  final String? searchTerm;


  PagedSearchRequest({this.page=1,  this.pageSize=10, this.searchTerm});

  Map<String, dynamic> toJson() => {
        'page': page,
        'pageSize': pageSize,
        if (searchTerm != null && searchTerm!.isNotEmpty) 'searchTerm': searchTerm,
      };
}

File: backend/TenderGo.Models/Requests/PagedSearchRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TenderGo.Models.Requests
{
    public class PagedSearchRequest
    {
        public int Page { get; set; } = 1;
        public int PageSize { get; set; } = 3;
    }
}
